src/main/scala/ch/epfl/bluebrain/nexus/iam/routes/PermissionsRoutes.scala

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.bluebrain.nexus.iam.routes
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import ch.epfl.bluebrain.nexus.iam.config.AppConfig.HttpConfig
+import ch.epfl.bluebrain.nexus.iam.config.AppConfig.tracing._
+import ch.epfl.bluebrain.nexus.iam.directives.AuthDirectives.authenticator
+import ch.epfl.bluebrain.nexus.iam.marshallers.instances._
+import ch.epfl.bluebrain.nexus.iam.permissions.{Permissions, Resource}
+import ch.epfl.bluebrain.nexus.iam.realms.Realms
+import ch.epfl.bluebrain.nexus.iam.routes.PermissionsRoutes.PatchPermissions
+import ch.epfl.bluebrain.nexus.iam.routes.PermissionsRoutes.PatchPermissions.{Append, Replace, Subtract}
+import ch.epfl.bluebrain.nexus.iam.types.ResourceF.resourceMetaEncoder
+import ch.epfl.bluebrain.nexus.iam.types.{Caller, Permission}
+import io.circe.syntax._
+import io.circe.{Decoder, DecodingFailure, Encoder, Json}
+import akka.http.javadsl.server.Rejections._
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+
+/**
+  * The permissions routes.
+  *
+  * @param permissions the permissions api
+  * @param realms      the realms api
+  */
+class PermissionsRoutes(permissions: Permissions[Task], realms: Realms[Task])(implicit http: HttpConfig) {
+
+  private implicit val resourceEncoder: Encoder[Resource] =
+    Encoder.encodeJson.contramap { r =>
+      resourceMetaEncoder.apply(r.discard) deepMerge Json.obj("permissions" -> Json.fromValues(r.value.map(_.asJson)))
+    }
+
+  def routes: Route =
+    (handleRejections(RejectionHandling()) & handleExceptions(ExceptionHandling())) {
+      path(http.prefix / "permissions") {
+        authenticateOAuth2Async("*", authenticator(realms)).withAnonymousUser(Caller.anonymous) { implicit caller =>
+          concat(
+            get {
+              parameter("rev".as[Long].?) { optRev =>
+                trace("fetchPermissions") {
+                  complete {
+                    optRev match {
+                      case Some(rev) => permissions.fetchAt(rev).runToFuture
+                      case None      => permissions.fetch.runToFuture
+                    }
+                  }
+                }
+              }
+            },
+            (put & parameter("rev".as[Long])) { rev =>
+              entity(as[PatchPermissions]) {
+                case Replace(set) =>
```

Comment:
In [the documentation](https://bluebrain.github.io/nexus/docs/api/iam/iam-permissions-api.html#replace-permissions) neither the create nor the replace require to have `"@type": "..."`. This was defined just for PATCH operations (Append and Subtract).

If we want to change that, we have to change it in the docs + in the ACLs routes too (in order for it to be consistent)